tests/conftest.py

```diff
@@ -25,6 +25,88 @@ def app():
     return app
 
 
+@pytest.fixture
+def dispatcher_local_mail_server(pytestconfig):
+    from aiosmtpd.controller import Controller
+
+    class CustomHandler:
+        async def handle_DATA(self, server, session, envelope):
+            try:
+                peer = session.peer
+                mail_from = envelope.mail_from
+                rcpt_tos = envelope.rcpt_tos
+                data = envelope.content
+                print(f"mail server: Receiving message from: {peer}")
+                print(f"mail server: Message addressed from: {mail_from}")
+                print(f"mail server: Message addressed to: {rcpt_tos}")
+                print(f"mail server: Message length : {len(data)}")
+            except:
+                return '500 Could not process your message'
+            return '250 OK'
+
+    handler = CustomHandler()
+
+    controller = Controller(handler, hostname='127.0.0.1', port=1025)
+    # Run the event loop in a separate thread
+    controller.start()
+
+    yield controller
+
+    print("will stop the mail server")
+    controller.stop()
+
+    # # with subprocess
```

Comment:
the asyncio version looks good to me, but you want to try also the subprocess one? maybe it is interesting, if so could you make another fixture out of it, with the same interface as the existing one? So we test it right away.

in principle subprocess was vital for server, since it is good to see the exact command used for running it.

Reply:
You can check now, a test with the subprocess is there, and the error is indeed generated

Reply:
@volodymyrss did you man age to reproduce the same error ?